shared/blob_utils.py

```
@@ -1,50 +1,138 @@
 BLOB_CONTAINER: str = "data"
 BLOB_NAME: str | None = "urls.json"
 
-async def load_urls() -> dict[str, list[dict[str, str]]] | None:
-    
+async def load_urls(key: str | None = None) -> list[dict[str, str | dict[str, str | int]]] | None:
     """
     Asynchronously loads and parses JSON data from a specific Azure Blob Storage blob.
     
-    Returns:
-        A dictionary containing the parsed JSON data if successful, or None if an error occurs during download or parsing.
+    If a key is provided, returns a list of items where the "key" field matches the given value. Returns None if no key is provided, or if an error occurs during download or parsing.
```

Pylint (3.3.7)

[convention] Line too long (182/100)

(C0301)